src/profile/tool/profile_background.py

```diff
@@ -0,0 +1,103 @@
+############################################################
+##
+## profile_background
+## generate a profile from prom query
+## 
+## ./python profile_background.py query_output_folder
+## e.g., ./python profile_background.py ../tests/prom_output
+##
+## input must be a query output of idle state
+##
+## profile saved in data/profile
+##  [source].json
+##   {component: {node_type: {min_watt: ,max_watt: } }}
+############################################################
+
+import sys
+import os
+
+src_path = os.path.join(os.path.dirname(__file__), '..', '..')
+train_path = os.path.join(os.path.dirname(__file__), '..', '..', 'train')
+
+sys.path.append(src_path)
+sys.path.append(train_path)
+
+from sklearn.preprocessing import StandardScaler
+
+from train import DefaultExtractor, node_info_column, component_to_col, PowerSourceMap
+
+import pandas as pd
+import json
+
+extractor = DefaultExtractor()
+
+min_watt_key = "min_watt"
+max_watt_key = "max_watt"
+
+profile_top_path = os.path.join(os.path.dirname(__file__), '..', '..', '..', 'resource', 'profiles')
+profile_path = os.path.join(profile_top_path, "profile")
+
+if not os.path.exists(profile_path):
+    os.mkdir(profile_path)
+
+def read_query_results(query_path):
+    results = dict()
+    metric_filenames = [ metric_filename for metric_filename in os.listdir(query_path) ]
+    for metric_filename in metric_filenames:
+        metric = metric_filename.replace(".csv", "")
+        filepath = os.path.join(query_path, metric_filename)
+        results[metric] = pd.read_csv(filepath)
+    return results
+
+def load_profile(source):
+    profile_filename = os.path.join(profile_path, source + ".json")
+    if not os.path.exists(profile_filename):
+        profile = dict()
+        for component in PowerSourceMap[source]:
+            profile[component] = dict()
+    else:
+        with open(profile_filename) as f:
+            profile = json.load(f)
+    return profile
+
+def save_profile(profile, source):
```

Comment:
I think we should look at better ways of saving a profile. Maybe we can treat profiles like users (set up a potential db)? This all depends once again on how we plan to manage contributors.

Reply:
That's true. We need the discussion about that to change this design.